### PR TITLE
TRT-2833: Add integration test tier for PostgreSQL-backed API methods

### PR DESCRIPTION
**Pull request by mstaeble**
## Summary
- Introduces a new integration test tier between unit tests (no DB) and e2e tests (full server) that validates SQL query logic against a real PostgreSQL instance using testcontainers-go
- Adds `make integration` target; 21 tests cover `ProwJobSimilarName`, `VariantReports`, and `JobReports` with edge cases (boundary timestamps, zero-denominator safety, multi-job aggregation, empty variants, cross-release isolation, filter/sort/limit, bug status filtering, PR retest calculations)
- Fixes four latent SQL bugs discovered through these tests:
  - `job_results` function returned `previous_failures` but Go struct expects `previous_fails` (column always scanned as zero)
  - All period-boundary queries used `BETWEEN` for both periods, double-counting runs at the boundary timestamp; previous period changed to half-open interval `[start, boundary)`
  - `job_results` open_bugs counted `COUNT(DISTINCT bug_jobs.bug_id)` which includes all associated bugs regardless of status; changed to `COUNT(DISTINCT bugs.id)` so closed/verified/modified/on_qa bugs are excluded via the existing LEFT JOIN filter
  - `job_results` LEFT JOIN to `bug_jobs`/`bugs` inside the `results` CTE caused row multiplication, inflating all aggregate counts (`current_runs`, `current_passes`, etc.) by the number of associated bugs; extracted bug counting into a separate `job_bugs` CTE joined at the outer SELECT level

## Test plan
- [x] `make lint` passes
- [x] `make test` passes
- [x] `make e2e` passes
- [x] `make integration` passes (21 tests)
- [ ] CI green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by mstaeble** [comment]
@coderabbitai full review

**Comment by mstaeble** [comment]
@coderabbitai full review

File: go.mod [hunk 1]
```diff
@@ -140,13 +173,12 @@ require (
 	go.opentelemetry.io/otel/sdk v1.43.0 // indirect
 	go.opentelemetry.io/otel/sdk/metric v1.43.0 // indirect
 	go.opentelemetry.io/otel/trace v1.43.0 // indirect
-	golang.org/x/crypto v0.50.0 // indirect
+	golang.org/x/crypto v0.51.0 // indirect
```
[the remark this answers is not in the document]
Comment (mstaeble) [reply go.mod:176]: The `golang.org/x/crypto` version here matches what's already on main. This PR didn't change it. The `testcontainers-go` dependency resolved against the existing module graph. A dependency bump should be done as a separate PR so it gets its own review and CI validation.

[the remark this answers is not in the document]
Comment (mstaeble) [reply go.mod:176]: No

[the remark this answers is not in the document]
Comment (mstaeble) [reply go.mod:176]: Created https://redhat.atlassian.net/browse/TRT-2844

**Comment by mstaeble** [comment]
@coderabbitai resolve

**Comment by mstaeble** [comment]
@coderabbitai full review
